src/dra/7A4D0.c

```diff
@@ -160,48 +160,49 @@ void func_8011A9D8(void) {
     }
 }
 
-Entity* CreateEntFactoryFromEntity(Entity* source, u32 flags, s32 arg2) {
-    Entity* entity;
-    s16 temp = arg2;
+Entity* CreateEntFactoryFromEntity(Entity* source, u32 flags, s32 arg2_raw) {
+    Entity* newFactory;
+    // Weird thing needed for callers to match
+    s16 arg2 = arg2_raw;
 
-    entity = GetFreeDraEntity(8, 16);
-    if (entity == NULL) {
+    newFactory = GetFreeDraEntity(8, 16);
+    if (newFactory == NULL) {
         return NULL;
     }
 
-    DestroyEntity(entity);
-    entity->entityId = E_UNK_1;
-    entity->ext.generic.unk8C.entityPtr = source;
-    entity->posX.val = source->posX.val;
-    entity->posY.val = source->posY.val;
-    entity->facingLeft = source->facingLeft;
-    entity->zPriority = source->zPriority;
-    entity->params = flags & 0xFFF;
+    DestroyEntity(newFactory);
+    newFactory->entityId = E_ENTITYFACTORY;
+    newFactory->ext.entFactory.parent = source;
+    newFactory->posX.val = source->posX.val;
+    newFactory->posY.val = source->posY.val;
+    newFactory->facingLeft = source->facingLeft;
+    newFactory->zPriority = source->zPriority;
+    newFactory->params = flags & 0xFFF;
     if (flags & 0x5000) {
-        entity->ext.generic.unkA8 = 0xE0;
+        newFactory->ext.entFactory.unkA8 = 0xE0;
     }
     if (flags & 0xA000) {
-        entity->ext.generic.unkA8 = 0xF0;
+        newFactory->ext.entFactory.unkA8 = 0xF0;
     }
-    entity->ext.generic.unkA0 = (flags >> 8) & 0xFF00;
-    entity->ext.generic.unk92 = temp;
+    newFactory->ext.entFactory.unkA0 = (flags >> 8) & 0xFF00;
+    newFactory->ext.entFactory.unk92 = arg2;
     if (source->flags & FLAG_UNK_10000) {
-        entity->flags |= FLAG_UNK_10000;
+        newFactory->flags |= FLAG_UNK_10000;
     }
     if (flags & 0x1000) {
```

Comment:
`FACTORY_FLAG_UNK_10000` and document the other spots?

Reply:
It looks like the 10000 is used in EntityEntFactory to just pass it down to its child entity so probably isn't worth having its own separate enum since the 10000 doesn't affect the behavior of the factory itself.

Ultimately every time an entity factory is used, there are 3 relevant entities. The one here called the "source", the factory itself, and the entity (or maybe entities) created by the factory.

In creating the factory here, we're saying if the source has 0x10000, give the factory 0x10000 as well, which is just to make it so that the children will also have 0x10000. That makes the child inherit the source's 0x10000 status, and just uses the factory as a go-between. No reason to treat it as different for the factory than for other entities.